src/esp/gfx/GenericDrawable.cpp

```diff
@@ -193,7 +233,16 @@ void GenericDrawable::draw(const Mn::Matrix4& transformationMatrix,
 }
 
 void GenericDrawable::updateShader() {
-  Mn::UnsignedInt lightCount = lightSetup_->size();
+  const Mn::UnsignedInt lightCount = lightSetup_->size();
+  const Mn::UnsignedInt jointCount =
+      skinData_ ? skinData_->skinData->skin->joints().size() : 0;
+  const Mn::UnsignedInt perVertexJointCount =
+      skinData_ ? skinData_->skinData->perVertexJointCount : 0;
+
+  if (skinData_) {
+    jointTransformations_ = Cr::Containers::Array<Mn::Matrix4>{
```

Comment:
Isn't this still a dynamic allocation? Can we just resize `jointTransformations_` here instead of creating and assigning a new array? The resize will be a no-op assuming the array was already the correct size from the previous call to `updateShader`.